src/kernel/include/inv.h

```diff
@@ -288,7 +288,7 @@ sinv_call(struct thread *thd, struct cap_sinv *sinvc, struct pt_regs *regs, stru
 		return;
 	}
 
-	if (unlikely(thd_invstk_push(thd, &sinvc->comp_info, ip, sp, cos_info))) {
+	if (unlikely(thd_invstk_push(thd, &sinvc->comp_info, ip, sp, fault_flag, cos_info))) {
```

Comment:
We are adding overhead for argument passing, and (eventually) for branches into the invocation path.  I'd like to understand the overhead of this.  Can you run the microbenchmark IPC cost test, and see if what the increase in cost is for the `sinv -> sret` path?